aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -203,5 +204,48 @@ Tensor& logical_xor_(Tensor& self, const Tensor& other) {
   return native::logical_xor_out(self, self, other);
 }
 
+Tensor& lt_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  TORCH_CHECK(result.scalar_type() == kBool,
+      "The output tensor of lt must be a bool.");
+  auto iter = TensorIterator::comparison_op(result, self, other, true);
+  lt_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor lt(const Tensor& self, const Tensor& other) {
+  Tensor result = at::empty({0}, self.options().dtype(kBool));
+  native::lt_out(result, self, other);
+  return result;
+}
+
+Tensor& lt_(Tensor& self, const Tensor& other) {
+  TORCH_CHECK(self.dtype() == other.dtype(),
```

Comment:
this check isn't so nice because we'll need to repeat it in each comparison function, but I don't have a better idea of what to do here.

Reply:
Changed